src/lava/magma/core/learning/learning_rule.py

```diff
@@ -0,0 +1,567 @@
+
+    """
+    pass
+
+
+class LoihiLearningRule(LearningRule):
+    """Encapsulation of learning-related information according to Loihi.
+
+    A LoihiLearningRule object has the following main objectives:
+    (1) Given string representations of learning rules (equations) describing
+    dynamics of the three synaptic variables (weight, delay, tag),
+    generate adequate ProductSeries representations and store them.
+
+    (2) Store other learning-related information such as:
+    impulse values by which to update traces upon spikes;
+    time constants by which to decay traces over time;
+    the length of the learning epoch;
+    a dict with dependencies as keys and the set of all traces appearing
+    with them in the specified learning rules as values;
+    the set of traces used by all specified learning rules.
+
+    From the user's perspective, a LoihiLearningRule object is to be used
+    as follows:
+    (1) Instantiate an implementation of LoihiLearningRule object with
+    learning rules given in string format for all three synaptic variables
+    (dw, dd, dt), as well as trace configuration parameters (impulse, decay)
+    for all available traces (x1, x2, y1), and the learning epoch length.
+
+    (2) The LoihiLearningRule object encapsulating learning-related information
+    is then passed to the Dense Process as instantiation argument.
+
+    (3) It will internally be used by ProcessModels to derive the operations
+    to be executed in the learning phase (Py and Nc).
+
+    Parameters
+    ----------
+
+    dw: str
+        ProductSeries representation of synaptic weight learning rule.
+    dd: str
+        ProductSeries representation of synaptic delay learning rule.
+    dt: str
+        ProductSeries representation of synaptic tag learning rule.
+    x1_impulse: float
+        Impulse by which x1 increases upon each pre-synaptic spike.
+    x1_tau: int
+        Time constant by which x1 trace decays exponentially over time.
+    x2_impulse: float
+        Impulse by which x2 increases upon each pre-synaptic spike.
+    x2_tau: int
+        Time constant by which x2 trace decays exponentially over time.
+    y1_impulse: float
+        Impulse by which y1 increases upon each post-synaptic spike.
+    y1_tau: int
+        Time constant by which y1 trace decays exponentially over time.
+    y2_impulse: float
+        Impulse by which y2 increases upon each post-synaptic spike.
+    y2_tau: int
+        Time constant by which y2 trace decays exponentially over time.
+    y3_impulse: float
+        Impulse by which y3 increases upon each post-synaptic spike.
+    y3_tau: int
+        Time constant by which y3 trace decays exponentially over time.
+    t_epoch: int
+        Duration of learning epoch.
+    rng_seed: int
+        Seed for the random number generators. If None, seed will be
+        set to current unix timestamp. Only used in fixed point
+        implementations.
```

Comment:
Not sure how this is used later on but if `rng_seed=None`, it may be best practice to generate a new numpy random number generator, rather than setting the seed to a time stamp.
https://numpy.org/doc/stable/reference/random/generator.html

Reply:
done